backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/question/QuestionService.java

```diff
@@ -262,6 +283,12 @@ public void deleteQuizQuestion(QuizQuestion quizQuestion) {
             backoff = @Backoff(delay = 5000))
     @Transactional(isolation = Isolation.READ_COMMITTED)
     public void deleteQuestion(Question question) {
+        QuestionSubmission questionSubmission = questionSubmissionRepository.findQuestionSubmissionByQuestionId(question.getId());
```

Comment:
need to discuss with teacher since the method is not mine and has different behaviour from removeQuestion. In the meantime I'll address the submission aspect and remove the ability to delete if one exists

Reply:
Yes, you can remove it and use the other one